### PR TITLE
Update dependencies to match latest @types/vscode

### PR DESCRIPTION
**Pull request by madkins23**
When I started trying to run the current version of Alive in VScode development mode I was getting 178 errors during compiles until I blew away my `node_modules` directory and executed:
```
npm i @types/vscode
```
This resulted in changes to `package.json` and `package-lock.json` which are included in this PR.

Just changing the `devDependencies` in `package.json` didn't get rid of the errors or the necessity to run the above `npm` command. I don't know if this was due to some misconfiguration on my system (Ubuntu 22.04, npm 8.5.1) or if there is something that can be changed in the code to avoid the issue.

Regardless, I've been carrying around these changes from branch to branch without submitting them so I figured I'd write up a PR and toss it on the pile.

### CODE REVIEW

**Comment by nobody-famous** [comment]
I'll need to look into this one some more. I'm not able to recreate the issue, but I'm on Windows right now. I'll try it again when I get a chance to on Linux.

I cloned a clean copy and it built without errors. It looks like it pulls the latest version of the vscode types, so I'm not sure why that would fix anything. Did nuking node_modules alone not fix it?

**Comment by nobody-famous** [comment]
Actually, I get the 178 errors if I try to compile before doing an initial npm -i, so the node_modules folder doesn't exist yet. Seems weird you'd get that with an existing node_modules directory.

**Comment by madkins23** [comment]
Yeah, that's what I thought. This fix doesn't remove that, either, it just makes these changes in the process. Seems like upgrading to a more recent version is a good idea ... if it doesn't break stuff.

**Comment by madkins23** [comment]
TLDR:  I tried to learn something about this issue and failed.

This was all done on the current master branch. Note that after the last two mergers I'm now seeing 177 errors instead of 178. No clue. Becomes important (?) later.

I thought I might avoid the changes to `package.json` and `package-lock.json` by blowing away `node_modules` and executing
```
npm i @types/vscode@1.50.0
```
to match what is requested in `package.json`.

Oddly enough what happened is that `package-lock.json` replaced the previous version `1.68.1` with the requested version `1.50.1`. It turns out that the `devDependencies` section of the current version of that file has `1.50.0` just like the same section in `package.json` and that was not changed. However there are currently sections further down showing
```
"node_modules/@types/vscode": {
            "version": "1.68.1",
```
and
```
        "@types/vscode": {
            "version": "1.68.1",
```
which were both changed. So there is a version mismatch in the `package-lock.json` file. Then there were 8 errors in the build step.

I then returned the code to current, cleared `node_modules`, and tried
```
npm i @types/vscode@1.68.1
```
In this case the `devDependencies` section of _both_ `package.json` and `package-lock.json` changed to the requested version. In this case there were no build errors.

Just for fun I cleared `node_modules` and tried building without the `npm` install using the "fixed" `package.json` and `package-lock.json` files. 177 errors.

Then I return to current state and just deleted the `package-json.test` file, figuring it would get rebuilt somehow. I cleared `node_modules` again, did _not_ install `@types/vscode`, and built the current code for 177 errors which is kind of expected (see note above). The `package-lock.json` file was not created.

With `package-lock.json` still absent I edited `package.json` to change the requested version to `1.68.1`. Build returned 177 errors. No `package-lock.json` was created.

A quick check shows that `package-lock.json` is created by `npm`. Since I'm skipping that step there is no expectation that it would be created.

I remain totally clueless. It does seem strange that the version in `package-lock.json` doesn't match `package.json` but I'm not sure what that means.

**Comment by nobody-famous** [comment]
package.json gives a range of versions that can be installed. For @types/vscode, it has "^1.50.0", which means anything above 1.50.0, but below 2.0.0.

package-lock.json has "1.68.0" because that's the version I had installed the last time I checked package-lock.json in.

When you run "npm install", it'll install whatever version of the modules is specified in package-lock.json. If there is no package-lock.json, or a module isn't in there, it pulls the latest based on the range in package.json and updates package-lock.json with whatever version it actually installed.

If you don't do the "npm install" step, it's expected to fail. Either at compile time because the dev dependecies aren't there or at runtime because the normal dependencies aren't there. If you have some of them installed globally, it might work without that step. I'm not sure, I don't install any npm modules globally.

**Comment by madkins23** [comment]
It looks like the tag for `1.50.0` [has been removed](https://www.npmjs.com/package/@types/vscode?activeTab=versions) or at least is not current. Per my experimentation yesterday the code compiles at `1.68.1` which makes sense as it was what you were using then.

What I was trying to understand was why the dependency wasn't loaded automatically by the build process. As near as I can tell that just doesn't happen. I think that the key is to run
```
npm install
```
in the build directory. This apparently reads `package.json` and downloads the required packages. Subsequently running
```
npm list
alive@0.3.20 /home/marc/work/lisp/alive
├── @types/node@14.18.21
├── @types/vscode@1.68.1
├── axios@0.21.4
├── node-stream-zip@1.15.0
├── ts-node@9.1.1
├── typescript@4.7.4
└── vscode-languageclient@7.0.0
```
shows version `1.68.1`. As a side effect this command increments the version count in the `package-lock.json` file. That change is in no way required and the file can be checked out again to remove the change and avoid redundant updates to the code tree.

So my misunderstanding was that the build process would load the required modules and that isn't true. When a developer starts working on (a fork of) the Alive code it is necessary to use `npm` to bootstrap the required modules. My ignorance of this process has now wasted a lot of your time and for that I apologize.

I would like to add the following items to the `README.md` (or  a separate `README-dev.md`):
* prior to running the code execute `npm install`
* checkout `package-lock.json` afterwards
This should be sufficient to prevent people like me from making a mess of things.

As for this PR, I think it can be rejected with prejudice. Again, sorry for dragging you through my learning process.

**Comment by nobody-famous** [comment]
OK, I'll go ahead and close this one. Having new people go through the process is good. I tend to know where the landmines are and will miss some things because of it. There's been plenty of issues I didn't run into because other people use things differently.

The npm install thing is standard node stuff. I think package-lock gets confusing and 99 times out of 100 is mostly useless. I tend to ignore it.

I agree that having a dev README would be a good idea. Not sure if it should be separate or if a section in the main README would be enough. Might depend on how complicated it gets. It should also include the tricks to work on the LSP. You can clone alive-lsp, set alive.lsp.install.path in the .vscode/settings.json for it to its own root directory, and then you can go into your alive repo, debug the extension, open the alive-lsp repo in the extension dev host, and edit the LSP that is being used to display the code you're editing.
